tests/pausing_test.go

```diff
@@ -363,30 +364,32 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 	Context("A long running process", func() {
 
+		grepSleepPid := func(expecter expect.Expecter) string {
+			res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
+				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
```

Comment:
shouldnt it start with Snd \n ?

Reply:
Snd \n + Exp prompt is done to make sure the buffer is clean (and actually a lot of our expecter flaky tests are because this cleanup failed). I think `tests.ExpectBatchWithValidatedSend` is safe enough and the extra cleanup in the beginning of the batch is not needed.

Reply:
it means that each previous batch must clean its output, unless you implement the snd random + exp random, we might encouter problems in the future in case previous exp spill chars (saw it in the past), but up to you

another corner this is prev batch shouldnt leave bg processes that might write stuff to stderr / stdout, but this is out of this PR anyway.

Reply:
Correct written batch is cleaning its output. But even if not, since we make sure the sent command appear before the expected value, even if there is leftover in the buffer, the test will work as expected and match would be found in the leftover.

2b. point in the [document](https://docs.google.com/document/d/16qGpswxJP9nDCprwNckXu5RHSUWF8lDFNmVEZT539aM/edit) suggests cleaning the buffer after each `BSnd`, `BExp` pair. We may have a follow-up PR doing it. Or as an alternative do a cleanup to our batches and make sure buffer is cleaned after each `BSnd` `BExp` pair. This can be achieved by using the new `Retcode` method in the `BExp`s. The method always waiting for the prompt.